app/components/menu/menu.js

```diff
@@ -90,14 +107,54 @@ const ButtonNavLink = ({ history, to, ...props }) => (
     {...props}
   />
 )
+const getNetworkStatus = async p2p => {
+  if (!p2p.networker || !(await isOnline())) return 'red'
+  if (!p2p.networker.swarm.holepunchable()) return 'yellow'
+  return 'green'
+}
 
-const Menu = ({ onFind }) => {
+const Menu = ({ p2p, onFind }) => {
   const history = useHistory()
   const { url: profileUrl } = useContext(ProfileContext)
+  const [networkStatus, setNetworkStatus] = useState('red')
+
+  useEffect(() => {
+    let unmounted = false
+    ;(async () => {
+      const check = async () => {
+        if (unmounted) return
+        const status = await getNetworkStatus(p2p)
+        if (unmounted) return
+        if (status !== networkStatus) setNetworkStatus(status)
+        setTimeout(check, 1000)
+      }
+      check()
+    })()
+    return () => {
+      unmounted = true
+    }
+  }, [networkStatus])
 
   return (
     <Container>
       <StyledLogo onClick={() => history.push('/')} />
+      <NetworkStatusContainer
+        title={
+          {
+            green: 'Connected',
+            yellow: 'Connected, but not holepunchable',
```

Comment:
Want something a bit more user-friendly for this. Open to other suggestions :)

```suggestion
            yellow: 'Connection limited (not holepunchable) - please check your firewall and network setup',
```